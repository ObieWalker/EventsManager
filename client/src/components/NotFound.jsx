import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PageNotFound from '../../assets/notFound.jpg';
/**
 *
 *
 * @class NotFound
 * @extends {Component}
 */
class NotFound extends Component {
  /**
   *
   *
   * @returns {object} not found  picture
   * @memberof NotFound
   */
  render() {
    return (
      <div className="bg-faded">
        <div className="not-found lead">
          <br />
          <br />
          <i className="large material-icons center">warning</i>
          {'    '}
          <img
            src={PageNotFound}
            style={{
              width: 400,
              height: 400,
              display: 'block',
              margin: 'auto',
              position: 'relative'
            }}
          />
          <center>
            <Link
              to="/"
              className={[
                'animated', 'bounceInUp', 'btn', 'btn-large'
              ].join(' ')}
            >
              Back to Home Page
            </Link>
          </center>
        </div>
      </div>
    );
  }
}

export default NotFound;
