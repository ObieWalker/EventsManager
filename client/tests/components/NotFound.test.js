import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import NotFound from '../../src/components/NotFound.jsx';

let mountedForm;
let props;
let wrapper;

const getComponent = () => {
  if (!mountedForm) {
    props = {};
    mountedForm = shallow(<NotFound {...props} />);
  }
  return mountedForm;
};

describe('Not Found Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });
  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper.find('Link').length).toBe(1);
    expect(wrapper.find('center').length).toBe(1);
    expect(wrapper.find('div').length).toBe(2);
    expect(wrapper.find('img').length).toBe(1);
  });
});
