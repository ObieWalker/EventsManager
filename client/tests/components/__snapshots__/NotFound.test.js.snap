// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Component renders component successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="not-found lead"
      >
        <br />
        <br />
        <i
          className="large material-icons center"
        >
          warning
        </i>
            
        <img
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "display": "block",
              "height": 400,
              "margin": "auto",
              "position": "relative",
              "width": 400,
            }
          }
        />
        <center>
          <Link
            className="animated bounceInUp btn btn-large"
            replace={false}
            to="/"
          >
            Back to Home Page
          </Link>
        </center>
      </div>,
      "className": "bg-faded",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <br />,
          <br />,
          <i
            className="large material-icons center"
          >
            warning
          </i>,
          "    ",
          <img
            src={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "display": "block",
                "height": 400,
                "margin": "auto",
                "position": "relative",
                "width": 400,
              }
            }
          />,
          <center>
            <Link
              className="animated bounceInUp btn btn-large"
              replace={false}
              to="/"
            >
              Back to Home Page
            </Link>
          </center>,
        ],
        "className": "not-found lead",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "warning",
            "className": "large material-icons center",
          },
          "ref": null,
          "rendered": "warning",
          "type": "i",
        },
        "    ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": Object {
              "process": [Function],
            },
            "style": Object {
              "display": "block",
              "height": 400,
              "margin": "auto",
              "position": "relative",
              "width": 400,
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="animated bounceInUp btn btn-large"
              replace={false}
              to="/"
            >
              Back to Home Page
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Back to Home Page",
              "className": "animated bounceInUp btn btn-large",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Back to Home Page",
            "type": [Function],
          },
          "type": "center",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="not-found lead"
        >
          <br />
          <br />
          <i
            className="large material-icons center"
          >
            warning
          </i>
              
          <img
            src={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "display": "block",
                "height": 400,
                "margin": "auto",
                "position": "relative",
                "width": 400,
              }
            }
          />
          <center>
            <Link
              className="animated bounceInUp btn btn-large"
              replace={false}
              to="/"
            >
              Back to Home Page
            </Link>
          </center>
        </div>,
        "className": "bg-faded",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <br />,
            <br />,
            <i
              className="large material-icons center"
            >
              warning
            </i>,
            "    ",
            <img
              src={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "display": "block",
                  "height": 400,
                  "margin": "auto",
                  "position": "relative",
                  "width": 400,
                }
              }
            />,
            <center>
              <Link
                className="animated bounceInUp btn btn-large"
                replace={false}
                to="/"
              >
                Back to Home Page
              </Link>
            </center>,
          ],
          "className": "not-found lead",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "warning",
              "className": "large material-icons center",
            },
            "ref": null,
            "rendered": "warning",
            "type": "i",
          },
          "    ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": Object {
                "process": [Function],
              },
              "style": Object {
                "display": "block",
                "height": 400,
                "margin": "auto",
                "position": "relative",
                "width": 400,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="animated bounceInUp btn btn-large"
                replace={false}
                to="/"
              >
                Back to Home Page
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Back to Home Page",
                "className": "animated bounceInUp btn btn-large",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Back to Home Page",
              "type": [Function],
            },
            "type": "center",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
